packages/jupyter-ai/jupyter_ai/document_loaders/directory.py

```diff
@@ -5,13 +5,28 @@
 from typing import List
 
 import dask
+from langchain.document_loaders import PyPDFLoader
 from langchain.schema import Document
 from langchain.text_splitter import TextSplitter
+from pypdf import PdfReader
+
+
+# Uses pypdf which is used by PyPDFLoader from langchain
+def pdf_to_text(path):
+    reader = PdfReader(path)
+    pages = reader.pages
+    text = ""
+    for page in pages:
+        text = text + "\n \n" + page.extract_text()
+    return text
 
 
 def path_to_doc(path):
     with open(str(path)) as f:
-        text = f.read()
+        if os.path.splitext(path)[1] == ".pdf":
```

Comment:
Some of our users are on Windows or macOS, where file systems are often configured to ignore casing of filenames. This would miss files ending in `.PDF`. Should we do this comparison in a case-insensitive way?

Reply:
See the code change for this
![Ensures all lower case comparison for PDF files](https://github.com/jupyterlab/jupyter-ai/assets/29005/93e2db73-1152-4e10-b8c5-ca5d5e560bef)